File: test/JsonModelGenTest/JsonModelGenTest/JsonModels/dependencies/JSONMorphoModel.m

//
//  JSONMorphoModel.m
//
//  Created by js2Model on 2014-11-05.
//  Copyright (c) 2014 Thomson Reuters. All rights reserved.
//

#import "JSONMorphoModel.h"
#import "TRJSONModelLoader.h"
#import "JSONModelSerialize.h"

@implementation JSONMorphoModelSchema

- (instancetype)init
{
    self = [super init];
    if (self) {

    }
    return self;
}

@end

static JSONMorphoModelSchema *JSONMorphoModelSchemaInstance;

@implementation JSONMorphoModel{
    NSMutableDictionary *_additionalProperties;
}

+(void)initialize {

    if( self == [JSONMorphoModel class] )
    {
        JSONMorphoModelSchemaInstance = [JSONMorphoModelSchema new];
    }
}

- (instancetype)init
{
    self = [super init];
    if (self) {
    	// custom intialization code
        _additionalProperties = [NSMutableDictionary new];
    }
    return self;
}


- (instancetype) initWithJSONData:(NSData *)data
                            error:(NSError* __autoreleasing *)error {
    self = [self init];
    if (self) {
        [TRJSONModelLoader load:self withJSONData:data error:error];
    }
    return self;
}

/** Parses JSON data and creates an Objective-C instance.

 @param cls Class type of top-most instance.
 @param filename Name of file with JSON data to be parsed.
 @param error Non-nil if any parsings errors occured.
 */
- (instancetype) initWithJSONFromFileNamed:(NSString *)filename
                                     error:(NSError* __autoreleasing *)error {

    self = [self init];
    if (self) {
        [TRJSONModelLoader load:self withJSONFromFileNamed:filename error:error];
    }
    return self;
}

- (JSONInstanceMeta *)objectForPropertyNamed:(NSString *)propertyName {

    return [JSONMorphoModelSchemaInstance objectForPropertyNamed:propertyName forInstance:self];
}

- (JSONInstanceMeta *)arrayForPropertyNamed:(NSString *)propertyName {
    return [JSONMorphoModelSchemaInstance arrayForPropertyNamed:propertyName forInstance:self];
}

- (void)setString:(NSString *)val forProperty:(NSString *)propertyName {
    [JSONMorphoModelSchemaInstance setString:val forProperty:propertyName forInstance:self];
}

- (void)setNumber:(NSNumber *)val forProperty:(NSString *)propertyName {
    [JSONMorphoModelSchemaInstance setNumber:val forProperty:propertyName forInstance:self];
}

- (void)setInteger:(NSNumber *)val forProperty:(NSString *)propertyName {
    [JSONMorphoModelSchemaInstance setInteger:val forProperty:propertyName forInstance:self];
}

- (void)setBoolean:(NSNumber *)val forProperty:(NSString *)propertyName {
    [JSONMorphoModelSchemaInstance setBoolean:val forProperty:propertyName forInstance:self];
}

- (void)setNullForProperty:(NSString *)propertyName {
    [JSONMorphoModelSchemaInstance setNullForProperty:propertyName forInstance:self];
}

+(JSONModelSchema *)modelSchema {
    return JSONMorphoModelSchemaInstance;
}

-(NSMutableDictionary*)additionalProperties {
    return _additionalProperties;
}

-(void)setValue:(id)value forAdditionalProperty:(NSString*)propertyName {
    [_additionalProperties setObject:value forKey:propertyName];
}

-(id)valueForAdditionalProperty:(NSString*)propertyName {
    return [_additionalProperties valueForKey:propertyName];
}

@end

File: test/js2ModelSwift/js2ModelSwift/JsonModels/dependencies/JsonMorphoModel.swift
//
//  JsonMorphoModel.swift
//  js2ModelSwift
//
//  Created by Kevin on 11/10/14.
//  Copyright (c) 2014 Thomson Reuters. All rights reserved.
//

import Foundation


//
//  TRPerson.m
//
//  Created by js2Model on 2014-11-05.
//  Copyright (c) 2014 Thomson Reuters. All rights reserved.
//

public class JsonMorphoModelSchema : JsonModelSchema<JsonMorphoModel> {
    
    init() { }
}

var JsonMorphoSchemaInstance: JsonMorphoModelSchema = JsonMorphoModelSchema();

class JsonMorphoModel : JsonModelSerialize {

    typealias ModelType = JsonMorphoModel
    typealias ModelSchemaType = JsonMorphoModelSchema;

    lazy var additionalProperties: [String: Any] = [String: Any]()

    required init() {
        
    }

    convenience required init(jsonData:[Byte]) {
        self.init()
    }
    
    convenience required init(filename:String) {
        self.init()
    }
    
    class func modelSchema() -> JsonMorphoModelSchema {
        return JsonMorphoSchemaInstance;
    }

    func set(string val:String, forProperty propertyName:String) {
        JsonMorphoSchemaInstance.set(string:val, forProperty:propertyName, forInstance:self);
    }
    
    func set(number val:Double, forProperty propertyName: String){
        JsonMorphoSchemaInstance.set(string:val, forProperty:propertyName, forInstance:self);
    }
 
    func set(integer val:Int, forProperty propetyName:String){
        JsonMorphoSchemaInstance.set(integer:val, forProperty:propertyName, forInstance:self);
    }
    
    func set(boolean val:Bool, forProperty propetyName:String){
        JsonMorphoSchemaInstance.set(boolean:val, forProperty:propertyName, forInstance:self);
    }
    
    func value(forAdditionalProperty propertyName:String) -> Any {
        return additionalProperties[propertyName]
    }
    
    func set(value:Any, forAdditionalProperty propertyName:String) {
        additionalProperties[propertyName] = value
    }
    
}
